game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/AaDefenseCombatValue.java

```diff
@@ -0,0 +1,95 @@
+package games.strategy.triplea.delegate.power.calculator;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.attachments.UnitSupportAttachment;
+import java.util.function.Predicate;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Value;
+
+/**
+ * Calculates defense strength and roll for AA/Targeted dice
+ *
+ * <p>This takes into account friendly support, and enemy support
+ */
+@Builder
+@Value
+@Getter(AccessLevel.NONE)
+class AaDefenseCombatValue implements CombatValue {
+
+  @NonNull GameData data;
+  @NonNull AvailableSupports friendlySupportTracker;
+  @NonNull AvailableSupports enemySupportTracker;
+
+  @Override
+  public StrengthOrRollCalculator getRoll() {
+    return new AaDefenseRoll(friendlySupportTracker, enemySupportTracker);
+  }
+
+  @Override
+  public StrengthOrRollCalculator getStrength() {
+    return new AaDefenseStrength(data, friendlySupportTracker, enemySupportTracker);
+  }
+
+  @Override
+  public boolean isDefending() {
+    return true;
+  }
+
+  @Override
+  public GameData getGameData() {
+    return data;
+  }
+
+  static class AaDefenseRoll extends StrengthOrRollCalculator {
+
+    AaDefenseRoll(final AvailableSupports friendlySupport, final AvailableSupports enemySupport) {
+      super(friendlySupport, enemySupport);
+    }
+
+    @Override
+    public int getValue(final Unit unit) {
+      final RollValue rollValue =
```

Comment:
nit, unnecessary intermediate variable, can just return with a single line.

It is a bit interesting we don't return a `RollValue` directly in this method, that would make for stronger typing. Any thoughts about doing that?

Reply:
My initial work returned the `RollValue`.  But the only place it is used needs an int and I didn't see the need to have the caller call `getValue` when it can be done here in this `getValue`.  Otherwise, the code would look like:
```
strengthCalculator.getValue(unit).getValue();
```